metrics_api/lib/opentelemetry/metrics/instrument/synchronous_instrument.rb

```diff
@@ -12,11 +12,34 @@ class SynchronousInstrument
         attr_reader :name, :unit, :description, :advice
 
         # @api private
-        def initialize(name, unit: nil, description: nil, advice: nil)
+        def initialize(name, unit: nil, description: nil, advice: nil, meter: nil)
           @name = name
           @unit = unit || ''
           @description = description || ''
           @advice = advice || {}
+
+          @meter = meter
+
+          @mutex = Mutex.new
+          @metric_streams = []
+        end
+
+        # @api private
+        def add_metric_stream(metric_stream)
+          @mutex.synchronize do
+            @metric_streams.push(metric_stream)
+            nil
+          end
+        end
+
+        private
+
+        def update(value, attributes)
+          @mutex.synchronize do
+            @metric_streams.each do |metric_stream|
+              metric_stream.update(value, attributes)
+            end
+          end
         end
```

Comment:
Moved `add_metric_reader` and `update` to the API Instrument class.

The logic to create a MetricStream is in the Meter